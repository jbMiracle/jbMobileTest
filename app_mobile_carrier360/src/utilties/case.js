
export const camelToUnderscore = text => text.replace(/([a-z])([A-Z])/g, (match, one, two) => `${one}_${two}`);

export const camelToSnake = text => camelToUnderscore(text).toLowerCase();
