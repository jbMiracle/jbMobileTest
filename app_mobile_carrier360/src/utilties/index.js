export * from './case';
export * from './validators';
export * from './formatters';
