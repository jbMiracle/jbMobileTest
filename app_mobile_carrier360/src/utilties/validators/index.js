export * from './emailValidator/emailValidator';
export * from './phoneNumberValidator/phoneNumberValidator';
