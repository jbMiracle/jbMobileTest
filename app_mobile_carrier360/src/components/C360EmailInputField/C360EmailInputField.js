import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { omit } from 'lodash';
import { C360InputField } from '../C360InputField/C360InputField';
import { isEmailAddressValid } from '../../utilties';

class C360EmailInputField extends PureComponent {
  validateEmail = (emailAddress) => {
    const { onValidationUpdate, onChangeText } = this.props;
    const hasValidEmail = isEmailAddressValid(emailAddress);

    if (onValidationUpdate) {
      onValidationUpdate(hasValidEmail);
    }

    if (onChangeText) {
      onChangeText(emailAddress);
    }
  };

  render() {
    const options = omit(this.props, ['onValidationUpdate', 'onChangeText', 'keyboardType']);

    return (
      <C360InputField
        { ...options }
        keyboardType='email-address'
        autoCapitalize='none'
        autoCorrect={ false }
        onChangeText={ this.validateEmail }
      />
    );
  }
}

C360EmailInputField.propTypes = {
  onValidationUpdate: PropTypes.func,
  onChangeText: PropTypes.func,
};

C360EmailInputField.defaultProps = {
  onValidationUpdate: undefined,
  onChangeText: undefined,
};

export default C360EmailInputField;
