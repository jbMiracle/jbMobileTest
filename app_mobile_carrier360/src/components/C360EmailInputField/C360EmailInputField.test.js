import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import C360EmailInputField from './C360EmailInputField';
import { isEmailAddressValid } from '../../utilties';

jest.mock('../../utilties', () => ({
  isEmailAddressValid: jest.fn(),
}));

describe('C360EmailInputField', () => {
  describe('performs the validateEmail operation in a pretend context', () => {
    isEmailAddressValid.mockImplementation(() => 'abc');

    const props = {
      onValidationUpdate: jest.fn(),
      onChangeText: jest.fn(),
    };

    const emailField = new C360EmailInputField(props);
    emailField.validateEmail('some text');

    it('should validate the inputted text and call onValidationUpdate with the result', () => {
      expect(props.onValidationUpdate.mock.calls[0][0]).toBe('abc');
    });

    it('should call onChangeText with the inputted text', () => {
      expect(props.onChangeText.mock.calls[0][0]).toBe('some text');
    });
  });

  it('can render an empty component', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<C360EmailInputField />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

