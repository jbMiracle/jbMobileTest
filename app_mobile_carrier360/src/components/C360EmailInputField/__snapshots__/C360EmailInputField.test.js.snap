// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`C360EmailInputField can render an empty component 1`] = `
<withAnimatable(Styled(C360InputField))
  animation={undefined}
  autoCapitalize="none"
  autoCorrect={false}
  delay={0}
  direction="normal"
  duration={undefined}
  easing={undefined}
  iterationCount={1}
  keyboardType="email-address"
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onChangeText={[Function]}
  style={undefined}
  transition={undefined}
  useNativeDriver={false}
/>
`;
